packages/itwinui-react/src/core/utils/components/Portal.tsx

```diff
@@ -31,11 +37,33 @@ export type PortalProps = {
  *   - if `portal` is set to true, renders into nearest ThemeContext.portalContainer
  *   - if `portal` is set to false, renders as-is without portal
  *   - otherwise renders into `portal.to` (can be an element or a function)
+ *     - If `to === undefined` (or `to() === undefined`), default behavior will be used (i.e. as if `portal = undefined`).
+ *     - E.g. `portal={{ to: () => document.querySelector('.may-not-exist') }}`.
  *
  * @private
  */
 export const Portal = (props: React.PropsWithChildren<PortalProps>) => {
-  const { portal = true, children } = props;
+  const { portal: portalProp = true, children } = props;
+
+  const portal = React.useMemo(() => {
```

Comment:
It would be much simpler to leave the `Portal` component unchanged, and only modify the internals of `usePortalTo`.

```diff
 const usePortalTo = (portal: NonNullable<PortalProps['portal']>) => {
   const themeInfo = React.useContext(ThemeContext);
+  const defaultPortal = themeInfo?.portalContainer ?? getDocument()?.body;
 
   if (typeof portal === 'boolean') {
-    return portal ? themeInfo?.portalContainer ?? getDocument()?.body : null;
+    return portal ? defaultPortal : null;
   }
 
-  return typeof portal.to === 'function' ? portal.to() : portal.to;
+  const portalTo = typeof portal.to === 'function' ? portal.to() : portal.to;
+  return portalTo ?? defaultPortal;
 };
```

Reply:
Thanks (https://github.com/iTwin/iTwinUI/pull/1786/commits/824d2eccf98f59cbc65d5579ba090035f8693a33). Yes, this cleaner. It also solved some unexpected issues I was having with the unit tests.

Reply:
small nit: rename `defaultPortal` → `defaultPortalTo`

Reply:
Sounds good. Changed. (https://github.com/iTwin/iTwinUI/pull/1786/commits/6fb572125e62515983c0d1a0fe3d23ca0e596a33)